src/utils/fp16_utils.cc

```diff
@@ -25,24 +25,14 @@
   Program grant you additional permission to convey the resulting work.
 */
 
-#include <cstdint>
+#include "utils/fp16_utils.h"
 #include <cstring>
 
-// Define NO_F16C to avoid the F16C intrinsics. Also disabled with NO_POPCNT
-// since it catches most processors without F16C instructions.
-
-#if defined(_M_IX86) || defined(_M_X64) || defined(__i386__) || \
-    defined(__x86_64__)
-#include <immintrin.h>
-#else
-#define NO_F16C
-#endif
-
 namespace lczero {
 
-uint16_t FP32toFP16(float f32) {
 #if defined(NO_POPCNT) || defined(NO_F16C) || \
     (defined(__GNUC__) && !defined(__F16C__))
+uint16_t FP32toFP16(float f32) {
   unsigned int x;
```

Comment:
Maybe put those implementations into the header too?

Reply:
Seemed like a lot, but then again we don't use it in too many places.